src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuildingGuards.java

```diff
@@ -0,0 +1,666 @@
+
+    /**
+     * Check if the guard should retrieve on low health.
+     *
+     * @return true if so.
+     */
+    public boolean shallRetrieveOnLowHealth()
+    {
+        return retrieveOnLowHealth;
+    }
+
+    /**
+     * Set if the guard should retrieve on low health.
+     *
+     * @param retrieveOnLowHealth state to set.
+     */
+    public void setRetrieveOnLowHealth(final boolean retrieveOnLowHealth)
+    {
+        this.retrieveOnLowHealth = retrieveOnLowHealth;
+        this.markDirty();
+    }
+
+    /**
+     * Check if the guard should patrol manually (Use defined values).
+     *
+     * @return true if so.
+     */
+    public boolean shallPatrolManually()
+    {
+        return patrolManually;
+    }
+
+    /**
+     * Set if the guard should patrol manually.
+     *
+     * @param patrolManually the state to set.
+     */
+    public void setPatrolManually(final boolean patrolManually)
+    {
+        this.patrolManually = patrolManually;
+        this.markDirty();
+    }
+
+    /**
+     * Getter of the position the gurd should guard.
+     *
+     * @return the blockPos position.
+     */
+    public BlockPos getGuardPos()
+    {
+        return guardPos;
+    }
+
+    /**
+     * Set if the jobs should be assigned manually.
+     *
+     * @param assignManually the state to set.
+     */
+    public void setAssignManually(final boolean assignManually)
+    {
+        this.assignManually = assignManually;
+        this.markDirty();
+    }
+
+    /**
+     * Possible job/AI.
+     */
+    public enum GuardJob
+    {
+        KNIGHT,
+        RANGER,
+    }
+
+    /**
+     * Possible tasks.
+     */
+    public enum Task
+    {
+        FOLLOW,
+        GUARD,
+        PATROL
+    }
+
+    /**
+     * The name of the guards's job.
+     *
+     * @return The name of the guards's job.
+     */
+    @NotNull
+    @Override
+    public String getJobName()
+    {
+        return GUARD_TOWER;
+    }
+
+    /**
+     * The client view for the baker building.
+     */
+    public static class View extends AbstractBuildingWorker.View
+    {
+        /**
+         * Assign the job manually, knight or ranger.
+         */
+        public boolean assignManually = false;
+
+        /**
+         * Retrieve the guard on low health.
+         */
+        public boolean retrieveOnLowHealth = false;
+
+        /**
+         * Patrol manually or automatically.
+         */
+        public boolean patrolManually = false;
+
+        /**
+         * The task of the guard, following the Task enum.
+         */
+        public Task task = Task.GUARD;
+
+        /**
+         * Position the guard should guard at.
+         */
+        public BlockPos guardPos = this.getID();
+
+        /**
+         * The job of the guard, following the GuarJob enum.
+         */
+        public GuardJob job = null;
+
+        /**
+         * The list of manual patrol targets.
+         */
+        public List<BlockPos> patrolTargets = new ArrayList<>();
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Make patrolTargets a static final constant or non-public and provide accessors if needed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AClassVariableVisibilityCheck)